core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java

```diff
@@ -104,7 +105,7 @@
       SparkConf conf,
       ShuffleWriteMetricsReporter writeMetrics) {
     // Use getSizeAsKb (not bytes) to maintain backwards compatibility if no units are provided
-    this.fileBufferSize = (int) conf.getSizeAsKb("spark.shuffle.file.buffer", "32k") * 1024;
+    this.fileBufferSize = (int) (long) conf.get(package$.MODULE$.SHUFFLE_FILE_BUFFER_SIZE()) * 1024;
```

Comment:
Why the 2 casts here?

Reply:
If we don't convert to `long`  first , it will encounter exception like this:
`Caused by: java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.Integer`

Reply:
Oh it's a Long object. You could call .intValue but I suppose its type is just Object at compile time. Ok.